packages/hebao_v2/contracts/base/libwallet/MetaTxLib.sol

```diff
@@ -26,20 +26,20 @@ library MetaTxLib
     using ERC20Lib      for Wallet;
 
     bytes32 public constant META_TX_TYPEHASH = keccak256(
-        "MetaTx(address relayer,address to,uint256 nonce,address gasToken,uint256 gasPrice,uint256 gasLimit,uint256 gasOverhead,bytes data)"
+        "MetaTx(address relayer,address to,uint256 timestamp,address gasToken,uint256 gasPrice,uint256 gasLimit,uint256 gasOverhead,bytes data)"
     );
 
     event MetaTxExecuted(
         address relayer,
-        uint    nonce,
+        bytes32 metaTxHash,
         bool    success,
         uint    gasUsed
     );
 
     struct MetaTx
     {
         address to;
-        uint    nonce;
+        uint    timestamp;
```

Comment:
Maybe make this `validUntil`, and check the timestamp against this so meta transactions cannot be used forever.